src/Common/SipHash.h

```diff
@@ -29,6 +31,61 @@ namespace ErrorCodes
 {
     extern const int LOGICAL_ERROR;
 }
+
+template <typename T>
+struct TransformToLittleEndian
+{
+    T operator()(const T & x) const noexcept
+    {
+        static_assert(std::is_scalar_v<T>);
```

Comment:
I'll see if I can be stricter. I hope nothing uses pointers for hashing, but enum values will still need to be transformed because they're simple integers.